storage/namespace.go

```diff
@@ -522,7 +551,23 @@ func (n *dbNamespace) Flush(
 	return res
 }
 
-func (n *dbNamespace) NeedsFlush(blockStart time.Time) bool {
+func (n *dbNamespace) NeedsFlush(start time.Time, end time.Time) bool {
```

Comment:
The method doc [here](https://github.com/m3db/m3db/blob/c93ecb1dc6806088f616b5ff8cf76960e9acba90/storage/types.go#L207-L209) describes inclusivity. The times don't need to be block aligned, i'll call that out too.